lib/utils/job-utils/racadm-tool.js

```diff
@@ -0,0 +1,221 @@
+        return self.getJobStatus(host, user, password, jobId)
+            .then(function(jobStatus){
+                if (jobStatus.status === 'Completed'){
+                    return jobStatus;
+                } else if (jobStatus.status === 'Failed'){
+                    throw {error: 'Job Failed during process',
+                        jobStatus: jobStatus};
+                } else if (jobStatus.status === 'Running'){
+                    if (retryCount < self.retries) {
+                        return Promise.delay(delay)
+                            .then(function () {
+                                retryCount += 1;
+                                delay = 2 * delay;
+                                return self.waitJobDone(host, user, password,
+                                    jobId, retryCount, delay);
+                            });
+                    } else {
+                        throw {error: 'Job Timeout',
+                            jobStatus: jobStatus};
+                    }
+                } else{
+                    throw {error: 'Job status is incorrect',
+                        jobStatus: jobStatus};
+                }
+            });
+    };
+
+    /**
+     * Returns a promise with the results or errors of enabling IPMI over lan
+     *
+     * @param {string} host
+     * @param {string} user
+     * @param {string} password
+     * @param {string} command
+     * @param {number} count
+     * @param {number} delay
+     */
+    RacadmTool.prototype.runAsynCommands = function(host, user, password, command, count, delay) {
+        var self = this;
+        return self.runCommand(host, user, password, command)
+            .then(function(consoleOutput){
+                return parser.getJobId(consoleOutput);
+            })
+            .then(function(jobId){
+                return self.waitJobDone(host, user, password, jobId, count, delay);
+            });
+    };
+
+    /**
+     * Returns a promise with the results or errors of setting BIOS configure
+     *
+     * @param {string} host
+     * @param {string} user
+     * @param {string} password
+     * @param {Object} cifsConfig - includes samba password, username and bios.xml file path
+     */
+    RacadmTool.prototype.setBiosConfig = function(host, user, password, cifsConfig) {
+        var command ='';
+        var cifsUser = cifsConfig.user,
+            cifsPassword = cifsConfig.password;
+        assert.string(cifsUser);
+        assert.string(cifsPassword);
+        var fileInfo = parser.getPathFilename(cifsConfig.filePath);
+
+        if (fileInfo.style === 'remote') {
+            command = "set -f " + fileInfo.name + " -t xml -u " +
+                cifsUser + " -p " + cifsPassword + " -l " + fileInfo.path;
+        } else if (fileInfo.style === 'local') {
+            command = "set -f " + fileInfo.path + "/" + fileInfo.name + " -t xml";
+        } else {
+            return Promise.reject({ error: 'XML file path is invalid'});
```

Comment:
Though I see some of our old code still reject a promise with a string, I think reject an Error instance should be a better practice, MDN also recommend to reject with an Error instance:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/reject

@RackHD/corecommitters  for comment.

    The static Promise.reject function returns a Promise that is rejected. For debugging purposes and selective error catching, it is useful to make reason an instanceof Error.